taskgraph/Task.py

```diff
@@ -1255,8 +1167,8 @@ def is_precalculated(self):
             self._reexecution_info['source_code_hash'],
             self._reexecution_info['other_arguments'],
             self._store_result,
-            # the x[2] is to only take the *hash* part of the 'file_stat'
-            str([x[2] for x in file_stat_list]))
+            # the x[1] is to only take the *hash* part of the 'file_stat'

```

Comment:
The "hash" part of this comment was a bit perplexing to me, since if we're using `"sizetimestamp"` as the `target_hash_algorithm`, this will end up being a string that isn't a digest at all.  I'm not sure if it's a bug in this comment specifically or a bug in the docstring of `_hash_file` (though that docstring specifically refers to a hash digest, which isn't strictly true), but what do you think?

Reply:
I'm okay with this since "sizetimestamp" is a hash too -- it changes one kind of data into a fingerprint. But also note it won't always be "sizetimestamp", it'll be whatever the value of `hash_algorithm` is. The point is `x[1]` is the hash component of the tuple that's returned from `_get_file_stats` and I otherwise thought it would be confusing if I left that `x[1]` hanging without comment.